src/System.Text.Json/tests/Utf8JsonWriterTests.cs

```diff
@@ -1228,6 +1307,17 @@ public void InvalidJsonPrimitive(bool formatted, bool skipValidation)
             {
                 Assert.Throws<InvalidOperationException>(() => jsonUtf8.WriteEndObject());
             }
+
+            jsonUtf8 = new Utf8JsonWriter(output, options);
+            jsonUtf8.WriteNumberValue(12345);
+            if (skipValidation)
+            {
+                jsonUtf8.WritePropertyName("test name");
+            }
+            else
+            {
+                Assert.Throws<InvalidOperationException>(() => jsonUtf8.WritePropertyName("test name"));
+            }
```

Comment:
and boolean, and null?

Reply:
There were diminishing returns, but will add some more cases since you brought it up :)

Reply:
I actually read these tests different than what they are.  I was expecting tests that show that WriteBoolean(propertyName, value) fail after a WritePropertyName, even when the container is an object.

I think got mislead by the test name (`InvalidJsonPrimitive`), since I thought it was "places it was invalid to write a primitive, in context")